Adding search by action and exports

mbayopanda opened the pull request:
This PR adds:
- Export option for CSV and Excel
- Search by type of action (delete or edit)

Close #5827

mbayopanda committed:
Adding search by action and exports

jniles commented on server/config/routes.js:
```diff
@@ -433,6 +433,7 @@ exports.configure = function configure(app) {
   app.get('/reports/finance/account_report', financeReports.reportAccounts.document);
   app.get('/reports/finance/account_report_multiple', financeReports.reportAccountsMultiple.document);
   app.get('/reports/finance/journal', financeReports.journal.postingReport);
+  app.get('/reports/finance/journal/log', financeReports.journal.journalLogReport);
```
:+1:

jniles reviewed (commented):
The CSV export works well 👍 

Could you do some updates to the XLSX export to make it more useful?  Users will want access to the information in the `value` section, but it currently isn't usable to them.

mbayopanda committed:
fix value for users

jniles reviewed (changes requested):
Have you tested this with the vanga database?  It crashes as is, but when I fixed the crash, it doesn't properly display all the information:

![image](https://user-images.githubusercontent.com/896472/128816449-f1f3d346-f178-41fe-a8fb-024b901afa13.png)

Since they are the ones who will use this information, let's try to fix this before merge.

mbayopanda committed:
Fix request parameters and timing

jniles committed:
fix: improve journal log

Ensures that we keep the hrRecord when deleting values, as well as
unparse the description appropriately.

jniles committed:
fix: replace invalid uuid in test data

Replaces an invalid UUID that could not be parsed in the test dataset.

jniles reviewed (approved):
LGTM.

Bors r+
